src/system/process.py

```diff
@@ -58,15 +58,33 @@ def terminate(self) -> int:
         logging.info(f"Process killed with exit code {self.process.returncode}")
 
         if self.stdout:
-            self.__stdout_data__ = open(self.stdout.name, 'r').read()
+            self.stdout.seek(0)
+            self.__stdout_data__ = self.stdout.read()
+            self.stdout.flush()
             self.stdout.close()
-            os.remove(self.stdout.name)
+            for proc in psutil.process_iter():
+                try:
+                    for item in proc.open_files():
+                        if self.stdout.name == item.path:
+                            raise Exception(f"stdout {item} is being used by process {proc}")
+                except Exception as err:
+                    logging.error(f"{err.args}")
```

Comment:
Do we want to retry closing here if the proc is still using the file?

Reply:
Not really. That's unexpected and we are supposed to get error I think.